### ISSUE TITLE
add YOLOV8

### ISSUE THREAD

**Issue by divyashreepathihalli** [keras-team/keras-hub#1760]
This task is part of the KerasHub project ( Consolidating all of KerasCV and KerasNLP models in to one place)
Models from KerasCV will be added to KerasNLP in KerasNLP infrstructure style. The development is now under the `keras-hub` branch in KerasNLP - https://github.com/keras-team/keras-nlp/tree/keras-hub

Steps to complete this Task
- Source code : https://github.com/keras-team/keras-cv/tree/master/keras_cv/src/models/object_detection/yolo_v8
- Destination : A model folder here https://github.com/keras-team/keras-nlp/tree/keras-hub/keras_nlp/src/models
- Example PR : https://github.com/keras-team/keras-nlp/pull/1737, please follow the style from here and take a look at the comments on the PR for any design clarifications
- copy the model over
- use KerasNLP test routines to update the tests
- for reformatting the code run `shell/api_gen.sh`, `shell/format.sh` and `shell/lint.sh`
- The backbone will be added as a model, any downstream task model will be under the same model folder.

Add either @divyashreepathihalli, @mattdangerw  or @SamanehSaadat for review of any PR.

**Comment by cosmo3769** [comment]
Hi @divyashreepathihalli, I would love to work on this. Please assign me. Thank you!

**Comment by shanky-kapoor** [comment]
@divyashreepathihalli Yolo_V8 is deeply integrated with bounding_box api of keras_cv. Also calls for a separate loss function CiOU [here](https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/models/object_detection/yolo_v8/yolo_v8_detector.py#L22) . 

I think this issue could be divided into two separate issues. 

Also when I was looking at code, it made sense to me to move bounding boxes and CiOU loss to Keras library rather than in keras_hub. But we can the continue discussion.

I am happy to pick this up.

**Comment by Sohaib-Ahmed21** [comment]
I have started working on this and will open PR soon. @divyashreepathihalli kindly assign me this issue. Thanks!

**Comment by cheachu** [comment]
@divyashreepathihalli , I was thinking to pick this up and start working :)

**Comment by Sohaib-Ahmed21** [comment]
> [@divyashreepathihalli](https://github.com/divyashreepathihalli) , I was thinking to pick this up and start working :)

I worked on it for a while but there is already a PR https://github.com/keras-team/keras-hub/pull/1899 open for this issue.

**Comment by divyashreepathihalli** [comment]
Hey all!! Because of license issues, we have decided to drop this.

**Comment by qllesudo** [comment]
@divyashreepathihalli What does that mean regarding license issues? Does it mean we cannot use YOLOv8 from KerasCV for commercial projects?

**Comment by sam31046** [comment]
> Hey all!! Because of license issues, we have decided to drop this.

@divyashreepathihalli I'm confused. Do you mean the PR #1899 will not be merged? Is there any license issue using KerasCV YOLOv8?
